1259 aws nodes

jdpinedac opened the pull request:
### What does this PR fix/introduce?

- Fix redaction and typos
- Update IaC devcontainer structure and handling
- Update and refactor of NOT_VPN to VPN_USE environment variable
- Modules reorganization and fix redaction
- Refactor OpenVPN configuration procedure
- Add IaC devcontainer troubleshooting.
- Add a VSCode devcontainer to build and test the documentation.

### Additional context

This Pull Request enhances the documentation for deploying a Casper node on AWS Cloud. Several updates were made to the existing documentation, including managing the VPN server, the included devcontainer for using infrastructure as code, among other elements, in addition to correcting wording and some typos. A devcontainer was also incorporated for the documentation itself.

### Checklist

- [x] Docs are successfully building - `yarn install && yarn run build`.
- [x] For new **internal** links I used *relative file paths* (with .md extension) - e.g. `../../faq/faq-general.md` - instead of introducing *absolute file path*, or *relative/absolute URL*.
- [x] All external links have been verified with `yarn run check:externals`.
- [x] My changes follow the [Casper docs style guidelines](https://docs.casper.network/resources/contribute-to-docs/).
- [x] All technical procedures have been tested (if you want help with this, mention it in [Reviewers](#reviewers)).
- [x] If structural changes are introduced (not just content changes), cross-broswer testing has been completed.

### Reviewers
@ipopescu

ipopescu commented:
@jdpinedac, thank you! I started the review.

ipopescu commented on .devcontainer/devcontainer.json:
```diff
@@ -12,7 +12,7 @@
 
 	// Use 'postCreateCommand' to run commands after the container is created.
-	"postCreateCommand": "sudo npm install -g textlint prettier tsdx @rollup/plugin-replace; yarn install",
+	"postCreateCommand": "sudo apt update; sudo apt install -y libnss3-dev libatk1.0-dev libatk-bridge2.0-dev libcups2-dev libdrm-dev libxkbcommon-dev libxcomposite-dev libxdamage-dev libxrandr-dev libgbm-dev libasound2-dev; sudo npm install -g textlint prettier tsdx @rollup/plugin-replace; yarn install",
```
FYI @bradjohnl

ipopescu commented:
@bradjohnl, please also do a technical review.

ipopescu commented:
Do not merge until the https://github.com/casper-ecosystem/casper-node-on-cloud/ repository is public.

ipopescu commented on source/docs/casper/operators/aws-nodes/4-modules.md:
```diff
@@ -87,129 +87,124 @@ A CloudWatch Agent obtains the following metrics every 30 seconds:
 | mem_used               | RAM usage in GB           |
```
Please update this table. Casper and the beginning of the description needs to be capitalized. Finally, to match the style, we need a colon at the end of each description. "Disk usage percentage." is probably better than "Disk usage in percentage."

| casper-node logs       | Casper service logs.       |
| casper-node error logs | Casper service error logs. |
| disk_total             | Disk total capacity.       |
| disk_used              | Disk usage in GB.          |
| disk_percent           | Disk usage percentage. |

ipopescu commented on source/docs/casper/operators/aws-nodes/4-modules.md:
```diff
@@ -87,129 +87,124 @@ A CloudWatch Agent obtains the following metrics every 30 seconds:
-### S3 Config
-
-The S3 config module creates an S3 bucket for storing additional configuration files such as `zip`,`.json`, or `.sh`.
+### Cloudwatch Logs
```
Please check the spelling to match the brand.
```suggestion
### CloudWatch Logs
```

ipopescu commented on source/docs/casper/operators/aws-nodes/4-modules.md:
```diff
@@ -52,13 +52,13 @@ The following table shows the alarms created by the module and their respective
 
```
Can we get an updated screenshot (email-notification.png) and remove the AWS highlighting? This one is over a year old.

ipopescu commented on source/docs/casper/operators/aws-nodes/4-modules.md:
```diff
@@ -8,21 +8,21 @@ This section describes AWS modules helpful for running and monitoring the node.
 ## Monitoring Modules
 
-This section describes all the modules related to monitoring the node. AWS provides various services to help operators monitor the node and Casper service status by creating alarms and having a visual representation in a dashboard, thus presenting the node's resources and capacity in real time.
+All modules pertinent to overseeing the node. AWS offers a suite of services enabling operators to observe both the node and Casper service status. These services allow the establishment of alarms and provide a visual representation on a dashboard, displaying the node's resources and capacity in real time.
```
```suggestion
Monitoring modules are pertinent to overseeing the node. AWS offers a suite of services enabling operators to observe the node and Casper service status. These services allow the establishment of alarms and provide a visual representation on a dashboard, displaying the node's resources and capacity in real-time.
```

ipopescu commented on source/docs/casper/operators/aws-nodes/4-modules.md:
```diff
@@ -8,21 +8,21 @@ This section describes AWS modules helpful for running and monitoring the node.
 ### CloudWatch Dashboard
 
-The IaC creates a dashboard to monitor all related resources. The dashboard includes metrics for connectivity status, blockchain information such as block height, block time, and era count, and node metrics such as CPU, memory, and disk usage metrics.
+The IaC establishes a dashboard to oversee all pertinent resources. This dashboard displays connectivity status, blockchain metrics like block height, block time, and era count, along with node metrics including CPU, memory, and disk usage.
```
```suggestion
The IaC establishes a dashboard to oversee all pertinent resources. This dashboard displays connectivity status, blockchain metrics like block height, block time, and era count, and node metrics including CPU, memory, and disk usage.
```

ipopescu commented on source/docs/casper/operators/aws-nodes/4-modules.md:
```suggestion
| `files/genSnapshot.sh`      | Bash file contains configurations to create a snapshot volume in EBS, with a cronjob performing weekly backups. |
```

ipopescu commented on source/docs/casper/operators/aws-nodes/4-modules.md:
```suggestion
| Zip Creation Module  | Module to compress information related to the synthetic canary. |
```

ipopescu commented on source/docs/casper/operators/aws-nodes/4-modules.md:
```suggestion
The following modules assist in data management.
```

ipopescu commented on source/docs/casper/operators/aws-nodes/4-modules.md:
Please use full sentences.
```suggestion
Network modules configure diverse network elements necessary for the node to function correctly.
```

ipopescu commented on source/docs/casper/operators/aws-nodes/4-modules.md:
```suggestion
The OpenVPN Service provides administrators with a private and secure connection to the node. This simple VPN is available for five administrators. To configure the VPN server, read the [OpenVPN guide](./6-open-vpn.md).
```

ipopescu reviewed (changes requested):
The new content is very clear and well written. Thank you, @jdpinedac!
I submitted a few yet minor change requests and re-requested a technical review from Luca. 
Let's not merge this PR with the main docs until the repository becomes public. Please remove the old wiki content from the repo as well. A few things to really note though:
- Use dev container not devcontainter, and introduce it initially. Please update all ocurrences of ‘devcontainer’, unless referring to the actual file.
- Please check the missing command in the open-vpn.md file.
- Add higher-res screenshots in the open-vpn.md file please.

jdpinedac commented on source/docs/casper/operators/aws-nodes/6-open-vpn.md:
````diff
@@ -31,26 +31,26 @@ Follow these steps to configure the VPN Server:
 
     ```bash
-    ssh -i "{keyname}.pem" openvpn@{Your-IPv4-Public-IP}
+    $ bash
````
Indeed, it is accurate and has been verified. While this shift may appear radical, it is particularly within the procedure that change is evident: there is no longer a need to use SSH. Instead, simply executing bash will automatically present the license and initiate the entire initial configuration process for OpenVPN.

ipopescu replied:
Great, thanks for clarifying @jdpinedac!

ipopescu commented:
@jdpinedac please let me know when you are ready for another review. Thanks!

jdpinedac commented on source/docs/casper/operators/aws-nodes/7-troubleshooting.md:
````diff
@@ -21,5 +21,16 @@ If the node restarts when synchronizing, usually every 30 minutes, and doesn't u
     ```
+
+## Devcontainer Troubleshooting
+
+- **Repeated container name:** A container with the same name may exist in another project. You can delete the previous container using the command docker container prune or docker container rm [containername].
+- **New image is not updating:** If you are changing the `Dockerfile`
+image or modifying the `entrypoint.sh` please restart VSCode or
+run the following command if you are not using VSCode:
+
+    ```bash
+    docker compose down; docker rmi {containername}:latest
+    ```
````
Thank you for the observation.

jdpinedac commented:
Hello @ipopescu,

I have reviewed and implemented the relevant suggestions you provided. I've also updated the screenshots that required better resolution. I encountered a peculiar issue: some of the suggestions are not visible in the PR conversation on GitHub, but they were visible using the GitHub Pull Requests extension in VSCode. I hope I didn’t overlook anything. Please let me know if I missed something or if anything is wrong.
Thank you so much!

ipopescu reviewed (approved):
Updates look great! Thank you, @jdpinedac! I will merge them to the `1259_aws_nodes`. 

FYI, @devendran-m, the documentation is ready to go. We can merge the `1259_aws_nodes` branch to `dev` when the [repository](https://github.com/casper-ecosystem/casper-node-on-cloud/) goes public.
Also, a reminder to remove the wiki content from the repository, because it is obsolete now.

ipopescu commented:
Updated the branch name to protect it: [feat-aws_nodes](https://github.com/casper-network/docs/tree/feat-aws_nodes).
